core/src/test/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/list/ListSwapMoveSelectorTest.java

```diff
@@ -333,9 +335,9 @@ void randomWithEntityValueRange() {
 
         assertCodesOfNeverEndingMoveSelector(moveSelector,
                 "2 {A[0]} <-> 3 {B[0]}",
-                "1 {A[1]} <-> 2 {A[0]}",
-                "3 {B[0]} <-> 2 {A[0]}",
-                "1 {A[1]} <-> 2 {A[0]}",
+                "No change",
+                "No change",
+                "No change",
```

Comment:
This is not ideal. 

We spent so much time making sure filtering is not necessary, only to end up in a place where we generate useless moves anyway? Sounds like we wasted time.